.ci/release/Jenkinsfile

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env groovy
+
+@Library('apm@current') _
+pipeline {
+  agent { label 'linux && immutable' }
+  environment {
+    REPO = 'apm-agent-java'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    NOTIFY_TO = credentials('notify-to')
+    DOCKERHUB_SECRET = 'secret/apm-team/ci/elastic-observability-dockerhub'
+    ELASTIC_DOCKER_SECRET = 'secret/apm-team/ci/docker-registry/prod'
+    NEXUS_SECRET = 'secret/apm-team/ci/nexus'
+    MAVEN_CONFIG = '-Dmaven.repo.local=.m2'
+    OPBEANS_REPO = 'opbeans-java'
+
+  }
+  options {
+    timeout(time: 3, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+
+  }
+  parameters {
+    string(name: 'MAVEN_CONFIG', defaultValue: '-B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn', description: 'Additional maven options.')
+    string(name: 'branch_specifier', defaultValue: 'master')
+  }
+
+  stages {
+    stage('Initializing'){
+      options { skipDefaultCheckout() }
+      environment {
+        HOME = "${env.WORKSPACE}"
+        JAVA_HOME = "${env.HUDSON_HOME}/.java/java10"
+        PATH = "${env.JAVA_HOME}/bin:${env.PATH}"
+        MAVEN_CONFIG = "${params.MAVEN_CONFIG} ${env.MAVEN_CONFIG}"
+      }
+      stages {
+        /**
+         Checkout the code
+        */
+        stage('Checkout') {
+          steps {
+            dir("${BASE_DIR}") {
+              // This is required to ensure that Maven is working with a real branch
+              sh(script: "git checkout -f {$branch_specifier}")
+            }
+          }
+        }
+      }
+    }
+    stage('Release') {
+      options { skipDefaultCheckout () }
+      stages{
+        // 1. Check to see if oss.sonatype.org is up. We throw an exception and fail the job if there is no response.
+        stage('Check oss.sonatype.org') {
+          steps {
+            // If this fails, an exception should be thrown and execution will halt
+            httpRequest(url: "https://oss.sonatype.org/#welcome")
```

Comment:
does the `#welcome` part of the URL matters here ?

Reply:
It doesn't matter. The root of the site redirects here so I just put it in to avoid the (slight) overhead of the redirect.

Reply:
For a future improvement: could we use this: https://status.maven.org/history.atom

Reply:
A while ago I looked into using the Atom feed for this kind of thing but it's fraught with problems. Unfortunately, it's basically just free-text and there is no clear way to determine whether a given entry represents an ongoing outage, a recovery, an announcement, or free puppies. 🐶 I have some ideas about how to do this kind of thing in the future with Github since they have enabled webhooks. My current plan is to get that working and then see if Sonatype is open to the possibility of using webhooks on their page as well.